### PR TITLE
Fix fstring errors in pyccel.ast

### PR DESCRIPTION
**Pull request by EmilyBourne**
Apply some more Pylint fixes. Namely:
- Fix fstring errors in pyccel.ast
- Fix several unused-import errors

Docstrings are also added to satisfy CI. Some deprecated code is removed:
- `get_function_from_ast` which was unused

### CODE REVIEW

File: pyccel/ast/utilities.py [hunk 1]
```diff
@@ -189,19 +189,6 @@ def builtin_import(expr):
 
     return []
 
-#==============================================================================
```
Comment (Farouk-Echaref) [inline pyccel/ast/utilities.py:192]: in what case was this method used before ?

**Review by Farouk-Echaref — APPROVED** [review]
looks good and clear, the use of fstring instead of format() makes sense

File: pyccel/ast/utilities.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/ast/utilities.py:192]: In no case that I could find. I have no idea what it was originally written for

**Review by yguclu — CHANGES_REQUESTED** [review]
That's very useful! I just have a couple of questions.

File: pyccel/ast/core.py [hunk 2]
```diff
@@ -163,10 +164,7 @@ def object(self):
         return self._obj
 
     def __repr__(self):
-        return '{0} as {1}'.format(str(self.name), str(self.target))
-
-    def __str__(self):
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/core.py:168]: This implementation of `__str__` duplicated the implementation of `__repr__`. By default if there is no `__str__` method defined and there is a `__repr__` method, the `__repr__` method is used to generate the string. Therefore the behaviour after this change is the same it just reduces the amount of duplicated code

File: pyccel/ast/core.py [hunk 3]
```diff
@@ -3782,6 +3779,28 @@ def __init__(self, source, target = None, ignore_at_print = False, mod = None):
...
+        """
+        Format a string passed to this file into a Pyccel object.
+
+        Format a string passed to this file into a Pyccel object or confirm
+        that it is already correctly formatted.
+
+        Parameters
+        ----------
+        i : obj
+            The object to be formatted.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/core.py:3791]: I think I meant to write `object` which is the thing from which you can make any class inherit `class MyClass(object)` but I wasn't sure what the correct type was here `Any` sounds good

[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/core.py]: Fixed ed166c0f4

File: pyccel/ast/numpy_wrapper.py [hunk 4]
```diff
@@ -60,17 +60,24 @@ class used to hold NumPy array objects in Python.
...
 def get_numpy_max_acceptable_version_file():
     """
+    Get the macro specifying the last acceptable numpy version.
+
     Get the macro specifying the last acceptable numpy version. If numpy is more
     recent than this then deprecation warnings are shown.
 
     The last acceptable numpy version is 1.19. If the current version is older
-    than this then the last acceptable numpy version is the current version
+    than this then the last acceptable numpy version is the current version.
```
[the remark this answers is not in the document]
Comment (EmilyBourne) [reply pyccel/ast/numpy_wrapper.py]: I can't put my finger on why but "newest acceptable" doesn't quite sound right to me. I have changed it to "most recent acceptable" and capitalised all instances of NumPy in comments in the file : 18ff50ff5

**Review by yguclu — APPROVED** [review]

**Pull request merged**